### DISCUSSION TITLE
Welcome to Customizer UI Discussions!

### DISCUSSION THREAD

**Discussion by BaronVonPerko** [BaronVonPerko/wpcui#45] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by bahiirwa** [comment]
Laurence is my name. I have an insatiable need for knowledge. So love to see the experimentation of the code.
WPCUI, offers me less headache trying to create new settings in customizer as a regular user. Hope to be a beta tester.

One question:

In version two since this is being build with react, will you be using the wrapped version of React (used in gutenberg)? Rather than import React from React? Keeps things updated and predictable.

**Reply by BaronVonPerko** [reply]
I'm glad you're enjoying the plugin so far!  I know it's quite limited in the current release.

I'm currently using the full React, but I'll check to see if the built-in version will work with what I'm doing so far.  May be using Portals soon so I'll need to make sure that it's compatible.  This is a great suggestion and would keep the plugin even smaller than it is now!

**Reply by bahiirwa** [reply]
> I'm currently using the full React, but I'll check to see if the built-in version will work with what I'm doing so far. May be using Portals soon so I'll need to make sure that it's compatible. This is a great suggestion and would keep the plugin even smaller than it is now!

https://developer.wordpress.org/block-editor/reference-guides/packages/packages-element/

**Comment by BaronVonPerko** [comment]
🚀 Took the time to finally move everything I've written so far to TypeScript.  I think this is going to help solve a lot of bugs going forward.  Check out the v2.0 branch to see what's complete so far 😀

I've created a v2.md file to explain what the v2 goals and features are.

**Comment by ciucacristi** [comment]
Hi! I would really like if your plugin would generate a php code to be able to use the wordpress customizer without this plugin installed. Is there any way to see the generated code behind the panels created or to keep the customizer created without your plugin active?

**Reply by BaronVonPerko** [reply]
Hi there, Ciuca!  At this time, the plugin does not generate any PHP code.  There are a series of loops that then run WordPress commands to create the sections and controls.  This is definitely something I have planned to do immediately after the 2.0 release.  I'm hoping to complete it soon, I've just been so busy with other things I haven't had time to work on it 😢
